tests/integration/pilot/ingress_test.go

```diff
@@ -273,7 +273,7 @@ spec:
 					HTTP: echo.HTTP{
 						Headers: headers.New().WithHost("bar.example.com").Build(),
 					},
-					Address: fmt.Sprintf("gateway.%s.svc.cluster.local", apps.Namespace.Name()),
+					Address: fmt.Sprintf("gateway-istio.%s.svc.cluster.local", apps.Namespace.Name()),
```

Comment:
@howardjohn is there a possibility that this type of fix is also causing all the `TestGatewayConformance` failures in the other integ tests?  I'm not sure how to debug those since they are all imported

Reply:
It wouldn't send to the cluster IP like this, but rather the IP reported in status of Gateway. Did you fix this part (status) yet?

```
Failed to assign to any requested addresses: hostname "httpbin-gateway.default.svc.cluster.local"
      not found'
```

Reply:
I thought I had fixed it with this [commit](https://github.com/istio/istio/pull/43103/commits/2a9caa2c2aaa115588172bcfb5d745a4cdea64af) since it resolved the docs example and got routing working.  I thought the `kubectl wait --for=condition=ready gtw httpbin-gateway` command returned but I guess not
```
$ k get gtw httpbin-gateway -o yaml
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"gateway.networking.k8s.io/v1beta1","kind":"Gateway","metadata":{"annotations":{},"name":"httpbin-gateway","namespace":"default"},"spec":{"gatewayClassName":"istio","listeners":[{"allowedRoutes":{"namespaces":{"from":"Same"}},"hostname":"httpbin.example.com","name":"http","port":80,"protocol":"HTTP"}]}}
  creationTimestamp: "2023-02-03T13:47:13Z"
  generation: 1
  name: httpbin-gateway
  namespace: default
  resourceVersion: "546375"
  uid: 9bc169a4-ab90-4ad4-9d43-7ae72c33eafc
spec:
  gatewayClassName: istio
  listeners:
  - allowedRoutes:
      namespaces:
        from: Same
    hostname: httpbin.example.com
    name: http
    port: 80
    protocol: HTTP
status:
  conditions:
  - lastTransitionTime: "2023-02-03T13:47:16Z"
    message: Deployed gateway to the cluster
    observedGeneration: 1
    reason: Accepted
    status: "True"
    type: Accepted
  - lastTransitionTime: "2023-02-03T13:47:16Z"
    message: Deployed gateway to the cluster
    observedGeneration: 1
    reason: ResourcesAvailable
    status: "True"
    type: Scheduled
  - lastTransitionTime: "2023-02-03T13:47:15Z"
    message: 'Assigned to service(s) httpbin-gateway-istio.default.svc.cluster.local:80,
      but failed to assign to all requested addresses: address pending for hostname
      "httpbin-gateway-istio.default.svc.cluster.local"'
    observedGeneration: 1
    reason: AddressNotAssigned
    status: "False"
    type: Ready
  listeners:
  - attachedRoutes: 0
    conditions:
    - lastTransitionTime: "2023-02-03T13:47:13Z"
      message: No errors found
      observedGeneration: 1
      reason: Accepted
      status: "True"
      type: Accepted
    - lastTransitionTime: "2023-02-03T13:47:13Z"
      message: No errors found
      observedGeneration: 1
      reason: NoConflicts
      status: "False"
      type: Conflicted
    - lastTransitionTime: "2023-02-03T13:47:13Z"
      message: No errors found
      observedGeneration: 1
      reason: Attached
      status: "False"
      type: Detached
    - lastTransitionTime: "2023-02-03T13:47:13Z"
      message: No errors found
      observedGeneration: 1
      reason: Programmed
      status: "True"
      type: Programmed
    - lastTransitionTime: "2023-02-03T13:47:13Z"
      message: No errors found
      observedGeneration: 1
      reason: Ready
      status: "True"
      type: Ready
    - lastTransitionTime: "2023-02-03T13:47:13Z"
      message: No errors found
      observedGeneration: 1
      reason: ResolvedRefs
      status: "True"
      type: ResolvedRefs
    name: http
    supportedKinds:
    - group: gateway.networking.k8s.io
      kind: HTTPRoute
```

Reply:
Hah - I knew I had checked for that already.  @howardjohn I think it may be timing out? While I was checking istiod debug API's to see if there was something still referencing the old naming scheme, I saw that there was an IP assigned Gateway in `/debug/configz`:
```
  {
    "kind": "Gateway",
    "apiVersion": "gateway.networking.k8s.io/v1beta1",
    "metadata": {
      "name": "httpbin-gateway",
      "namespace": "default",
      "resourceVersion": "546743",
      "creationTimestamp": "2023-02-03T13:47:13Z",
      "annotations": {
        "kubectl.kubernetes.io/last-applied-configuration": "{\"apiVersion\":\"gateway.networking.k8s.io/v1beta1\",\"kind\":\"Gateway\",\"metadata\":{\"annotations\":{},\"name\":\"httpbin-gateway\",\"namespace\":\"default\"},\"spec\":{\"gatewayClassName\":\"istio\",\"listeners\":[{\"allowedRoutes\":{\"namespaces\":{\"from\":\"Same\"}},\"hostname\":\"httpbin.example.com\",\"name\":\"http\",\"port\":80,\"protocol\":\"HTTP\"}]}}\n"
      }
    },
    "spec": {
      "gatewayClassName": "istio",
      "listeners": [
        {
          "allowedRoutes": {
            "namespaces": {
              "from": "Same"
            }
          },
          "hostname": "httpbin.example.com",
          "name": "http",
          "port": 80,
          "protocol": "HTTP"
        }
      ]
    },
    "status": {
      "addresses": [
        {
          "type": "IPAddress",
          "value": "34.132.98.142"
        }
      ],
      "conditions": [
        {
          "lastTransitionTime": "2023-02-03T13:47:58Z",
          "message": "Deployed gateway to the cluster",
          "observedGeneration": 1,
          "reason": "Accepted",
          "status": "True",
          "type": "Accepted"
        },
        {
          "lastTransitionTime": "2023-02-03T13:47:58Z",
          "message": "Deployed gateway to the cluster",
          "observedGeneration": 1,
          "reason": "ResourcesAvailable",
          "status": "True",
          "type": "Scheduled"
        },
        {
          "lastTransitionTime": "2023-02-03T13:47:57Z",
          "message": "Gateway valid, assigned to service(s) httpbin-gateway-istio.default.svc.cluster.local:80",
          "observedGeneration": 1,
          "reason": "ListenersValid",
          "status": "True",
          "type": "Ready"
        }
```
and sure enough, I checked the ready command again and it returned:
```
kubectl wait --for=condition=ready gtw httpbin-gateway
gateway.gateway.networking.k8s.io/httpbin-gateway condition met
```

Reply:
hmm, takes anywhere from 10s to almost a minute on my GKE env, the `TestGatewayConformance/GatewayObservedGenerationBump/observedGeneration_should_increment` has enough retries there, but not sure how long it is waiting since there are no timestamps.  It's the same test that's failed on all of them so it doesn't seem like a timing flake

Reply:
kind should create the LB ~instantly so that wouldn't explain CI. I can try to poke around and see if anything stands out